### PR TITLE
Add set records as viewed in data load task

### PR DESCRIPTION
**Pull request by davidjray**
Thanks @prescod  for the help with the unit tests.

# Critical Changes

# Changes
- Added an new option`set_recently_viewed` to the `load_dataset` task that sets newly inserted data as recently viewed

# Issues Closed

### CODE REVIEW

**Review by prescod — APPROVED** [review]

**Comment by prescod** [comment]
I helped write this so it wouldn't hurt to have independent eyes on it. It's small.

File: cumulusci/tasks/bulkdata/load.py [hunk 1]
```diff
@@ -101,6 +104,9 @@ def _init_options(self, kwargs):
         self.options["drop_missing_schema"] = process_bool_arg(
             self.options.get("drop_missing_schema") or False
         )
+        self.options["set_recently_viewed"] = process_bool_arg(
+            self.options.get("set_recently_viewed") or False
```
Comment (davisagli) [inline cumulusci/tasks/bulkdata/load.py:108]: From slack it sounds like the decision was to change this default to True, right?

File: cumulusci/tasks/bulkdata/load.py [hunk 2]
```diff
@@ -631,3 +639,27 @@ def _generate_contact_id_map_for_person_accounts(
...
+        if custom_objects:
+            for record in self.sf.query_all(
+                "SELECT SObjectName FROM TabDefinition WHERE IsCustom = true AND SObjectName IN ('{}')".format(
+                    "','".join(sorted(custom_objects))
+                )
+            )["records"]:
+                object_names.add(record["SobjectName"])
+
+        for mapped_item in sorted(object_names):
+            self.sf.query_all(f"SELECT Id FROM {mapped_item} LIMIT 1000 FOR VIEW")
```
Comment (davisagli) [inline cumulusci/tasks/bulkdata/load.py:665]: There's nothing here to make sure that we're specifically selecting records that were just inserted, is there? Should we do a reverse sort by creation date? Should we limit it to the number of items that were inserted?

File: cumulusci/tasks/bulkdata/load.py (hunk 1 above)
Comment (davidjray) [reply cumulusci/tasks/bulkdata/load.py:108]: @davisagli yessir, I am looking at making the change now, it has some repercussions on some unit tests.

**Review by prescod — APPROVED** [review]

**Review by davisagli — APPROVED** [review]

**Pull request merged**